models/userSchemas.ts

```diff
@@ -0,0 +1,44 @@
+const mongoose = require('mongoose');
+
+enum userRole {
+    Admin = 0,
+    Writer = 1,
+    Reader = 2
+};
+
+enum contentType {
+    Text = 0,
+    Image = 1,
+    Video = 2
+};
+
+enum contentColumn {
+    Left = 0, 
+    Middle = 1,
+    Right = 2
+};
+
+const userSchema = new mongoose.Schema({
+    userID: Number,
+    name: String,
+    email: String,
+    profilePicture: String, // String -- maybe imgr
+    role: userRole,
+    submissions: Array<Number>
```

Comment:
`submissions` should be an array of submissions for the various nodes and not an array of number. To do this, you would have to create a model and an interface with each schema, which you can read more about it [here](https://mongoosejs.com/docs/models.html)